_python/main/admin.py

```diff
@@ -428,12 +429,52 @@ def live_annotations_count(self, obj):
 
 ## Users
 
-class UserAdmin(BaseAdmin):
-    readonly_fields = ['created_at', 'updated_at', 'display_name', 'last_request_at', 'last_login_at', 'login_count', 'login']
+class UserAddForm(forms.ModelForm):
+    """
+        Override DjangoUserAdmin.add_form so "add user" uses a standard form, except for setting random user password
+        on creation so the recover-password feature will work.
+    """
+    class Meta:
+        model = User
+        fields = '__all__'
+
+    def save(self, commit=True):
+        user = super().save(commit=False)
+        self.instance.set_password(User.objects.make_random_password(length=20))
+        if commit:
+            user.save()
+        return user
+
+
+class UserAdmin(BaseAdmin, DjangoUserAdmin):
+    filter_horizontal = ('roles', 'verified_professor')
```

Comment:
Whoops! I think this is a little broken. Django says it can't have a horizonal filter on `'verified_professor'` since that isn't a many-to-many field.
```
django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:

ERRORS:
<class 'main.admin.UserAdmin'>: (admin.E020) The value of 'filter_horizontal[1]' must be a many-to-many field.
```